patches/tModLoader/Terraria/ModLoader/ModNPC.cs

```diff
@@ -114,6 +114,20 @@ public abstract class ModNPC : ModTexturedType
 		/// </summary>
 		public virtual ModNPC Clone() => (ModNPC)MemberwiseClone();
 
+		/// <summary>
+		/// Whether or not this NPC will spawn with a custom name like a town NPC. Make sure to override <see cref="TownNPCName"/> in order to determine what name will be selected.
+		/// Returning true will force a name to be rolled regardless of vanilla behavior. Returning null will roll a name by normal vanilla specifications.
+		/// Returning false will prevent any name being rolled, regardless of vanilla behavior. Returns null by default.
+		/// </summary>
+		/// <returns></returns>
+		public virtual bool? SpawnsWithCustomName => null;
+
+		/// <summary>
+		/// Whether or not this NPC will act like a town NPC in terms of AI, animations, and attacks; but not in other regards, like the bone merchant in vanilla. Returns false by default.
+		/// </summary>
+		/// <returns> </returns>
+		public virtual bool ActsLikeATownNPC => false;
+
```

Comment:
I don't see much reason for these 2 to not just be NPCID.Sets.* entries. Especially since you deemed GlobalNPC entries unnecessary.